include/astarte_device_sdk/formatter.hpp

```diff
@@ -0,0 +1,436 @@
+// (C) Copyright 2025, SECO Mind Srl
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef ASTARTE_FORMATTER_H
+#define ASTARTE_FORMATTER_H
+
+#include <chrono>
+#include <cstddef>
+#include <type_traits>
+
+// TODO(rgallor): stop using spdlog formatter once C++20 will become the minimu required version
+#if (__cplusplus >= 202002L) && (__has_include(<format>))
+#include <format>
+#define NS_FORMAT std
```

Comment:
I would have this macro start with `ASTARTE`. To avoid name clashing with user defined macros.